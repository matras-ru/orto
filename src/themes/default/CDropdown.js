export const wrapperBase = 'relative';
export const dropdownBase = 'absolute z-50 top-full min-w-full mt-0-4 bg-white overflow-hidden';
export const dropdownVariantPrimary = 'shadow-example rounded';
export const dropdownVariantSecondary = 'shadow-example rounded-lg';
export const dropdownPlacementLeft = 'left-0';
export const dropdownPlacementRight = 'right-0';

export default {
    wrapperBase,
    dropdownBase,
    dropdownVariantPrimary,
    dropdownVariantSecondary,
    dropdownPlacementLeft,
    dropdownPlacementRight
};
