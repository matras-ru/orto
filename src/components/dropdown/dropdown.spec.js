/* eslint-env jest */
import { mount, createLocalVue } from '@vue/test-utils';
import * as vClickOutside from 'v-click-outside-x';
import CDropdown from './dropdown';

describe('Dropdown basic', () => {
    const localVue = new createLocalVue();
    localVue.use(vClickOutside);

    it('is rendered', () => {
        const wrapper = mount(CDropdown, {
            localVue
        });

        expect(wrapper.element.tagName).toEqual('DIV');
        expect(wrapper.classes('relative')).toBe(true);

        wrapper.destroy();
    });

    it('dropdown isShow', () => {
        const wrapper = mount(CDropdown, {
            localVue,
            data() {
                return {
                    isShow: true
                };
            }
        });

        const dropdown = wrapper.findComponent({ ref: 'dropdown' });

        expect(dropdown.element.tagName).toEqual('DIV');
        expect(dropdown.classes().sort()).toEqual(
            'absolute z-50 top-full min-w-full left-0 mt-0-4 bg-white overflow-hidden shadow-example rounded'
                .split(' ')
                .sort()
        );
    });

    it('placement', () => {
        const wrapper = mount(CDropdown, {
            localVue,

            propsData: {
                placement: 'right'
            },

            data() {
                return {
                    isShow: true
                };
            }
        });

        expect(wrapper.vm.placement).toBe('right');
        const dropdown = wrapper.findComponent({ ref: 'dropdown' });

        expect(dropdown.element.tagName).toEqual('DIV');
        expect(dropdown.classes()).not.toContain('left-0');
        expect(dropdown.classes()).toContain('right-0');
    });
});

describe('Dropdown slots', () => {
    const localVue = new createLocalVue();
    localVue.use(vClickOutside);

    it('holder', () => {
        const wrapper = mount(CDropdown, {
            localVue,
            scopedSlots: {
                holder: function () {
                    return this.$createElement('div', 'Click me');
                }
            }
        });

        const holder = wrapper.find('div div');
        expect(holder.element.tagName).toEqual('DIV');

        wrapper.destroy();
    });

    it('holder - toggle', async () => {
        const wrapper = mount(CDropdown, {
            localVue,
            scopedSlots: {
                holder: function ({ toggle }) {
                    return this.$createElement(
                        'div',
                        {
                            on: {
                                click: () => {
                                    toggle();
                                }
                            }
                        },
                        'Click me'
                    );
                }
            }
        });

        const holder = wrapper.find('div div');
        let dropdown;
        expect(wrapper.vm.isShow).toBe(false);
        dropdown = wrapper.findComponent({ ref: 'dropdown' });
        expect(dropdown.exists()).toBe(false);

        holder.trigger('click');
        await wrapper.vm.$nextTick();

        expect(wrapper.vm.isShow).toBe(true);
        dropdown = wrapper.findComponent({ ref: 'dropdown' });
        expect(dropdown.exists()).toBe(true);
        holder.trigger('click');
        await wrapper.vm.$nextTick();

        expect(wrapper.vm.isShow).toBe(false);
        expect(wrapper.vm.$refs.dropdown).toEqual(undefined);

        wrapper.destroy();
    });

    it('dropdown - toggle', async () => {
        const wrapper = mount(CDropdown, {
            localVue,
            data() {
                return {
                    isShow: true
                };
            },
            scopedSlots: {
                dropdown: function ({ toggle }) {
                    return this.$createElement(
                        'div',
                        {
                            attrs: {
                                id: 'innerContent'
                            },
                            on: {
                                click: () => {
                                    toggle();
                                }
                            }
                        },
                        'Click me'
                    );
                }
            }
        });

        const innerContent = wrapper.find('#innerContent');

        expect(wrapper.vm.isShow).toBe(true);

        innerContent.trigger('click');
        await wrapper.vm.$nextTick();

        expect(wrapper.vm.isShow).toBe(false);

        wrapper.destroy();
    });
});
