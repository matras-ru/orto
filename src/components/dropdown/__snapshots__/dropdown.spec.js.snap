// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown basic dropdown isShow 1`] = `
<div class="relative">
  <div class="absolute z-50 top-full min-w-full mt-0-4 bg-white overflow-hidden shadow-secondary rounded left-0"></div>
</div>
`;

exports[`Dropdown basic placement 1`] = `
<div class="relative">
  <div class="absolute z-50 top-full min-w-full mt-0-4 bg-white overflow-hidden shadow-secondary rounded right-0"></div>
</div>
`;

exports[`Dropdown slots holder 1`] = `
<div class="relative">
  <div>Click me</div>
</div>
`;
